### PR TITLE
Vectorize Combine Quaternion node

### PR DESCRIPTION
**Pull request by harisreedhar**
Vectorized Combine Quaternion node

### CODE REVIEW

**Review by OmarEmaraDev — CHANGES_REQUESTED** [review]
Looks good. But you need to normalize the quaternions.

**Comment by harisreedhar** [comment]
> Looks good. But you need to normalize the quaternions.

Should i write Quaternion Normalize function on [/math/quaternion.pyx](https://github.com/JacquesLucke/animation_nodes/blob/master/animation_nodes/math/quaternion.pyx)?

**Comment by OmarEmaraDev** [comment]
Yes.

**Review by harisreedhar — COMMENTED** [review]
Normalized quaternion output

File: animation_nodes/math/quaternion.pyx [hunk 1]
```diff
@@ -47,3 +47,14 @@ cdef void quatFromAxisAngle(Quaternion *q, Vector3 *axis, float angle):
...
+@cython.cdivision(True)
+cdef void quaternionNormalize_InPlace(Quaternion *q):
+    cdef float length = sqrt(q.w * q.w + q.x * q.x + q.y * q.y + q.z * q.z)
+    if length != 0:
+        q.w /= length
+        q.x /= length
+        q.y /= length
+        q.z /= length
+    else:
+        q.w = q.x = q.y = q.z = 0
```
Comment (OmarEmaraDev) [inline animation_nodes/math/quaternion.pyx:60]: When the length is zero, those values are already zero, so this line is redundant.

**Review by OmarEmaraDev — CHANGES_REQUESTED** [review]

File: animation_nodes/math/quaternion.pyx (hunk 1 above)
Comment (harisreedhar) [reply animation_nodes/math/quaternion.pyx]: removed else statement

**Review by OmarEmaraDev — APPROVED** [review]
Thanks!

**Pull request merged**